R/estimatr_lm_lin.R

```diff
@@ -253,14 +253,25 @@ lm_lin <- function(formula,
   # Center and interact variables
   # ----------
 
+  # Initialize as non-demeaned
+  demeaned_covars <-
+    design_matrix[
+      ,
+      setdiff(colnames(design_matrix), c(design_mat_treatment, "(Intercept)")),
+      drop = FALSE
+    ]
+
+  # Choose what to center on!
+  if (is.numeric(weights)) {
+    center <- apply(demeaned_covars, 2, weighted.mean, weights)
+  } else {
+    center <- TRUE
```

Comment:
```
center <- colMeans(...)
```

or simliar - generally don't set a variable to different types in different branches.